app/src/app/store/mapRenderSubs.ts

```diff
@@ -34,6 +35,20 @@ export const getRenderSubscriptions = (
   useMapStore: typeof _useMapStore,
   useHoverStore: typeof _useHoverStore
 ) => {
+  const addColorSchemeSub = useMapStore.subscribe<MapStore['colorScheme']>(
+    state => state.colorScheme,
+    colorScheme => {
+      removeZoneMetaLayers();
+      debouncedAddZoneMetaLayers({});
+      const {getMapRef, mapDocument, addVisibleLayerIds} = useMapStore.getState();
+      if (mapDocument) {
+        addBlockLayers(getMapRef(), mapDocument);
```

Comment:
H: This is a very expensive way to update the colors -- changing layer styles will be substantially easier once we merge in the react-map-gl refactor but I'd suggest just updating style properties instead of fully adding and removing layers